lib/BackbeatProducer.js

```diff
@@ -128,13 +128,17 @@ class BackbeatProducer extends EventEmitter {
                 cb(errors.InternalError);
             });
         }
-        const messages = this._partition === undefined ?
-            entries.map(item => new KeyedMessage(item.key, item.message)) :
-            entries.map(item => item.message);
-        const payload = [{
-            topic: this._topic,
-            messages,
-        }];
+        const payload = this._partition === undefined ?
+                  entries.map(item => ({
+                      topic: this._topic,
```

Comment:
We don't necessarily need to set topic or key for the message as they are optional. Topic is set at producer level. Was is impacting the KeyedMessage?

Reply:
The key is needed here for the message to be sent to the proper partition with keyed partitioning. As for the topic, it was there before so I didn't change this. Actually I don't see where else it's set so i think it's necessary to provide it here.